### ISSUE TITLE
Running exporter's command without root privileges

### ISSUE THREAD

**Issue by gunhu** [tynany/frr_exporter#28]
Hi @tynany,

I would love to be able to run your prometheus exporter without root privileges when vtysh command is needed.

This PR [#27](https://github.com/tynany/frr_exporter/pull/27) implement the described logic.

When running without root privileges, sudo is invoked and we need this kind of sudo stanza to make it work:

```
username ALL=(ALL) NOPASSWD: /usr/bin/vtysh -c show *
```

Regards,
gunhu.

**Comment by tynany** [comment]
I believe your requirement can be achieved without modifying frr_exporter, but instead adding the non-root user running frr_exporter to the `frrvty` group.

Taken from the [FRR vtsh documentation](https://docs.frrouting.org/en/latest/vtysh.html):
> Users using vtysh still need to have appropriate access to the daemons’ VTY sockets, usually by being member of the frrvty group

The below systemd file and group membership works with the current version of frr_exporter for a non-root user (in this example the non-root user is frr_exporter).

Systemd file using a non-root user:

```
[Unit]
Description=FRR Exporter

[Service]
User=frr_exporter
ExecStart=/usr/sbin/frr_exporter 
Restart=always

[Install]
WantedBy=multi-user.target
```

And the relevant line in `/etc/group`:
```
frrvty:x:100:frr,frr_exporter
```

Let me know what you think.

**Comment by karlism** [comment]
I'm running `frr_exporter` as `frr` user and it is working just fine. If you are running it on RedHat based OS you might want to check out https://github.com/lest/prometheus-rpm as package from that repository will install and configure `frr_exporter` to run with `frr` user.

**Comment by gunhu** [comment]
Indeed but I don't really want a user to perform any action on FRRouting instance.

Only `show` commands are needed by your exporter.

**Comment by fdomain** [comment]
Wouldn't be also interesting to use the [`--user` vtysh option] (https://docs.frrouting.org/en/latest/vtysh.html#permissions-and-setup-requirements) for all vtysh calls made by the exporter? It forbids access to any configuration command.

**Comment by gunhu** [comment]
@fdomain I think it's better to restrict it with sudo because the documentation is saying `No guarantees of security are provided for this option and under no circumstances should this option be used to provide any semblance of security or read-only access to FRR.`

**Comment by gunhu** [comment]
As requested by @tynany I implemented a sudo flag on the exporter to avoid breaking current implementation of the exporter. I hope it will suit your needs too.

**Comment by tynany** [comment]
@gunhu thanks for implementing this! Just confirming this has been released in [v0.1.14](https://github.com/tynany/frr_exporter/releases/tag/v0.2.14).
